theia-extensions/viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -11,25 +11,33 @@ export class TraceServerServiceImpl implements TraceServerConfigService {
     private server?: ChildProcess;
     private port?: number;
 
-    async startTraceServer(path?: string, port?: number): Promise<string> {
+    async startTraceServer(path?: string, cliArgs?: string): Promise<string> {
         if (path === undefined) {
             throw new Error('no Trace Server path specified');
-        } else if (port === undefined) {
+        } else if (cliArgs === undefined) {
+            throw new Error('no Trace Server cli arguments specified');
+        }
+
+        const port = cliArgs.match(/(?<=port=)\d+/g);
+        // eslint-disable-next-line no-null/no-null
+        if (port === null) {
             throw new Error('no Trace Server port specified');
         } else if (this.isServerRunning(this.server)) {
-            if (this.port === port) {
+            if (this.port === parseInt(port[0])) {
                 return SUCCESS;
             } else {
                 throw new Error('the Trace Server is already running on a different port');
             }
         }
-        const server = spawn(path, ['-vmargs', `-Dtraceserver.port=${port}`]);
+
+        const argsArray: Array<string> | undefined = cliArgs?.split(' ');
+        const server = spawn(path, argsArray);
         if (server.pid === undefined) {
             // When pid is undefined it usually means we're about to get an error.
             return new Promise<never>((_, reject) => server.once('error', reject));
         }
         this.server = server;
-        this.port = port;
+        this.port = parseInt(port[0]);
```

Comment:
instead of saving the port you should save the cliArgs and replace the port with it in the file (line 23, line 54, line 63).